code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1311,3 +1314,16 @@
 			locked = 1
 		notify_ai(1)
 		. = 1
+
+
+/mob/living/silicon/robot/proc/security_level(security_level = SEC_LEVEL_GREEN)
+	if(security_level == SEC_LEVEL_RED || security_level == SEC_LEVEL_DELTA)
+		if(designation == "Peacekeeper")
+			var/obj/item/borg/upgrade/reset/re = new
+			re.action(src)
+			pick_module("Security")
+	else
+		if(designation == "Security")
```

Comment:
I'm not too sure about using the existing module picking system anyway, since it locks the cyborg into place for a few seconds while it plays the module switch animation. When a peacekeeper goes into harmbaton mode I think it should just stay a peacekeeper but with a slightly altered sprite (weapon arms extended and some red lights to make it look more intimidating), and access to secborg modules.